cognee/infrastructure/databases/graph/neptune_driver/adapter.py

```
@@ -0,0 +1,1445 @@
+            result = await self.query(query, params)
+
+            # Format edges as EdgeData tuples: (source_id, target_id, relationship_name, properties)
+            edges = [self._convert_relationship_to_edge(record) for record in result]
+
+            logger.debug(f"Retrieved {len(edges)} edges for node: {node_id}")
+            return edges
+
+        except Exception as e:
+            error_msg = format_neptune_error(e)
+            logger.error(f"Failed to get edges for node {node_id}: {error_msg}")
+            raise Exception(f"Failed to get edges: {error_msg}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)